src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -485,33 +486,46 @@ export const handleFormAuthRedirect = [
 
 /**
  * NOTE: This is exported only for testing
- * Logs user out of SP / CP By deleting cookie
+ * Logs user out of SP / CP / MyInfo / SGID by deleting cookie
  * @param authType type of authentication
  *
  * @returns 200 with success message when user logs out successfully
  * @returns 400 if authType is invalid
  */
 export const _handleSpcpLogout: ControllerHandler<
-  { authType: AuthType.SP | AuthType.CP | AuthType.SGID },
+  { authType: AuthType.SP | AuthType.CP | AuthType.MyInfo | AuthType.SGID },
   PublicFormAuthLogoutDto
 > = (req, res) => {
   const { authType } = req.params
 
-  return res
-    .clearCookie(JwtName[authType])
-    .status(200)
-    .json({ message: 'Successfully logged out.' })
+  switch (authType) {
+    case AuthType.MyInfo:
+      return res
+        .clearCookie(MYINFO_COOKIE_NAME)
+        .status(200)
+        .json({ message: 'Successfully logged out.' })
```

Comment:
Should we create a new service function in PublicFormService to retrieve the cookie name so the controller does not need to know these business logic? For example:

```ts
// public-form.service.ts
const getCookieNameByAuthType = (authType: AuthType) => {
  switch (authType):
    case AuthType.MyInfo:
      return MYINFO_COOKIE_NAME
   ...
}

// public-form.controller.ts
// here
return res.clearCookie(getCookieNameByAuthType(authType).status(200).json({...})
```

Cleans up the repetitive nature

Reply:
nice, thanks! edited